### ISSUE TITLE
Patch cannot be determined when only-new-isses: true and push tag

### ISSUE THREAD

**Issue by fredbi** [golangci/golangci-lint-action#1246]
### Welcome

- [x] Yes, I understand that the GitHub action repository is not the repository of golangci-lint itself.
- [x] Yes, I've searched similar issues on GitHub and didn't find any.
- [x] Yes, I've included all information below (version, config, etc).

### Description of the problem

Good morning

I am pushing a tag to cut a release:

> git tag v0.32.1
> git push origin v0.32.1

The lint job fires on CI and attempts to determine the changes.

The action retrieves a wrong "before" commit from the github context 
[here](https://github.com/golangci/golangci-lint-action/blob/5286ed6a0851c6869b5bce32518201fef4ca2173/src/patch.ts#L92).

The before SHA posted to the github API is 0.
```  
   status: 404,
    response: {
      url: 'https://api.github.com/repos/go-swagger/go-swagger/compare/0000000000000000000000000000000000000000...aece8f0e60b8195764c9c42fb615d82c42a19cd8',
``` 

Now golangci-lint runs like:
``` 
golangci-lint run --nw-from-rev 0
``` 

and spews out older linting issues that have remained there for some time.

> This is related to, but different from #996

> CI output: https://github.com/go-swagger/go-swagger/actions/runs/15530304923/job/43717653580

I might be related to this report about the behavior of th github context: https://github.com/orgs/community/discussions/116616

### Version of golangci-lint

v2.6.1

### Version of the GitHub Action

v7

### Workflow file

<details>

```yml
on:
  push:
    tags:
      - v*
    branches:
      - master
    paths-ignore:
      - docs/*
      - hack/hugo/*
      - .github/workflows/update-doc.yaml

permissions:
  contents: write
  pull-requests: read

jobs:
  lint:
    name: lint
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-go@v5
        with:
          go-version: stable
          check-latest: true
          cache: true
      - name: golangci-lint
        uses: golangci/golangci-lint-action@v7
        with:
          args: --verbose
          only-new-issues: true
          skip-cache: true
          version: latest

```

</details>

### Golangci-lint configuration

<details>

```yml
<add your file content here>
```

</details>

### Go version

1.24.4

### Code example or link to a public repository

<details>

```go
// add your code here
```

</details>

**Comment by ldez** [comment]
Hello,

I recommend adding a condition on the golangci-lint step:

```yml
jobs:
  lint:
    name: lint
    runs-on: ubuntu-latest
    if: !startsWith(github.event.ref, 'refs/tags/v')
    steps:
```

The GitHub push event, when triggered on a tag, doesn't have a before SHA. This is expected as a tag doesn't have a parent.

**Comment by fredbi** [comment]
Yeah this what I figured out.

**Comment by ldez** [comment]
> However, this has been working before,

There may be a GitHub API change, but I don't think so.

> Parent or not parent, you can always associate a tag with its commit and get its parent.

The event is `push`, not `tag`: the "parent" (base commit) of a push is not always the previous commit.
